### ISSUE TITLE
[feature] one week one month one year

### ISSUE THREAD

**Issue by dawadam** [valamidev/candlestick-convert#14]
Hi, thanks for work.
What about one year, month or week ?
It will be nice.

**Comment by valamidev** [comment]
Hi,

good idea thanks, I will extend with this feature.

**Comment by dawadam** [comment]
great !
take into account that the first day of the week may be different depending on the country

**Comment by valamidev** [comment]
Yes, that is why it will be implemented like allowing a custom interval function(timestamp: number) => [open,close], it will allow to use any custom logic for dynamic intervals.

**Comment by valamidev** [comment]
It is available in them 6.0.0 version:

```
batchCandleArrayCustomInterval(candleData: OHLCV[], intervalFunction: IntervalFunction, includeOpenCandle = false)

export type IntervalFunction = (timeStamp: number) => number;
```

The IntervalFunction allow you to add any timestamp based logic for batching.
